### PR TITLE
search: aggregator merges file match results in report

### PR DESCRIPTION
**Pull request by keegancsmith**
This is a refactoring that moves the merge logic from the invidual
functions to the new report function. Additionally we implement true
merge logic, rather than assumptions based on if we have a symbol vs
filematch result.

There is one potential downside. We now allocate an intermediate
"[]SearchResultResolver" slice. In practice I believe this won't be that
big of an issue since the size of the interface "SearchResultResolver"
is tiny compared to the size of a "FileMatchResolver".

### CODE REVIEW

File: cmd/frontend/graphqlbackend/textsearch.go [hunk 1]
```diff
@@ -123,6 +123,7 @@ func (fm *FileMatchResolver) path() string {
 // counts and limit.
 func (fm *FileMatchResolver) appendMatches(src *FileMatchResolver) {
 	fm.JLineMatches = append(fm.JLineMatches, src.JLineMatches...)
+	fm.symbols = append(fm.symbols, src.symbols...)
```
Comment (keegancsmith) [inline cmd/frontend/graphqlbackend/textsearch.go:126]: cc @rvantonder

File: cmd/frontend/graphqlbackend/search_results.go [hunk 2]
```diff
@@ -1688,12 +1689,6 @@ func (a *aggregator) doFilePathSearch(ctx context.Context, args *search.TextPara
 		tr.Finish()
 	}()
 	fileResults, fileCommon, err := searchFilesInRepos(ctx, args)
-	a.mu.Lock()
-	defer a.mu.Unlock()
-	// Timeouts are reported through searchResultsCommon so don't report an error for them
```
Comment (stefanhengl) [inline cmd/frontend/graphqlbackend/search_results.go:1693]: Should we preserve this hint? Seems like something one can easily overlook.

**Review by stefanhengl — APPROVED** [review]
Nice!

File: cmd/frontend/graphqlbackend/textsearch.go (hunk 1 above)
Comment (rvantonder) [reply cmd/frontend/graphqlbackend/textsearch.go:126]: looks good to me, is there any specific reason you're calling out this line? :-)

**Review by rvantonder — APPROVED** [review]

File: cmd/frontend/graphqlbackend/search_results.go (hunk 2 above)
Comment (keegancsmith) [reply cmd/frontend/graphqlbackend/search_results.go:1693]: will add it in

**Pull request merged**